aws-janitor/resources/target_groups.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// Clean-up ELBV2 target groups.
+
+type TargetGroups struct{}
+
+func (TargetGroups) MarkAndSweep(opts Options, set *Set) error {
+	logger := logrus.WithField("options", opts)
+	svc := elbv2.New(opts.Session, aws.NewConfig().WithRegion(opts.Region))
+	var targetGroups []*targetGroup
+	tgTags := make(map[string]Tags)
+
+	pageFunc := func(page *elbv2.DescribeTargetGroupsOutput, _ bool) bool {
+		for _, tg := range page.TargetGroups {
+			a := &targetGroup{
+				arn: aws.StringValue(tg.TargetGroupArn),
+			}
+			targetGroups = append(targetGroups, a)
+			tgTags[a.ARN()] = nil
+		}
+		return true
+	}
+
+	if err := svc.DescribeTargetGroupsPages(&elbv2.DescribeTargetGroupsInput{}, pageFunc); err != nil {
+		return err
+	}
+
+	fetchTagErr := incrementalFetchTags(tgTags, 20, func(tgArns []*string) error {
```

Comment:
I'm not sure whether do we need to use incrementalFetchTags here, maybe we can just use fromEC2Tags instead?

Reply:
TargetGroups don't seem to have Tags on them, so we can't use fromEC2Tags here, and ELBv2's DescribeTags is limited to 20 ARNs per request, so we do need to use incrementalFetchTags here.